spec/design/skel/Intermediate_H.thy

```diff
@@ -48,9 +48,8 @@ defs createNewCaps_def:
 "createNewCaps t regionBase numObjects userSize dev \<equiv>
     (case toAPIType t of
           Some TCBObject \<Rightarrow> (do
-            addrs \<leftarrow> createObjects regionBase numObjects (injectKO (makeObject ::tcb)) 0;
             curdom \<leftarrow> curDomain;
-            mapM_x (\<lambda>tptr. threadSet (tcbDomain_update (\<lambda>_. curdom)) tptr) addrs;
+            addrs \<leftarrow> createObjects regionBase numObjects (injectKO ((makeObject ::tcb)\<lparr>tcbDomain := curdom\<rparr>)) 0;
```

Comment:
interesting, but it seems more consistent now to go via createObjects

Reply:
I'm not sure what you mean by this.

Reply:
I mean I approve of your change, thumbs up, etc